WhateverGreen/kern_igfx.cpp

```diff
@@ -248,3 +349,493 @@ bool IGFX::wrapAcceleratorStart(IOService *that, IOService *provider) {
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTableOffsetIndexSlice)
+					platformInformationList[i].fBTTableOffsetIndexSlice = callbackIGFX->framebufferPatch.fBTTableOffsetIndexSlice;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTableOffsetIndexNormal)
+					platformInformationList[i].fBTTableOffsetIndexNormal = callbackIGFX->framebufferPatch.fBTTableOffsetIndexNormal;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTableOffsetIndexHDMI)
+					platformInformationList[i].fBTTableOffsetIndexHDMI = callbackIGFX->framebufferPatch.fBTTableOffsetIndexHDMI;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFCameliaVersion)
+					platformInformationList[i].cameliaVersion = callbackIGFX->framebufferPatch.cameliaVersion;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFNumTransactionsThreshold)
+					platformInformationList[i].fNumTransactionsThreshold = callbackIGFX->framebufferPatch.fNumTransactionsThreshold;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFVideoTurboFreq)
+					platformInformationList[i].fVideoTurboFreq = callbackIGFX->framebufferPatch.fVideoTurboFreq;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTArraySliceAddr)
+					platformInformationList[i].fBTTArraySliceAddr = callbackIGFX->framebufferPatch.fBTTArraySliceAddr;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTArrayNormalAddr)
+					platformInformationList[i].fBTTArrayNormalAddr = callbackIGFX->framebufferPatch.fBTTArrayNormalAddr;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFBTTArrayHDMIAddr)
+					platformInformationList[i].fBTTArrayHDMIAddr = callbackIGFX->framebufferPatch.fBTTArrayHDMIAddr;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFSliceCount)
+					platformInformationList[i].fSliceCount = callbackIGFX->framebufferPatch.fSliceCount;
+				
+				if (callbackIGFX->framebufferPatchFlags.bits.FPFEuCount)
+					platformInformationList[i].fEuCount = callbackIGFX->framebufferPatch.fEuCount;
+			}
+		}
+	}
+	
+	for (int i=0; i<MAX_FRAMEBUFFER_PATCH_COUNT; i++) {
+		if (callbackIGFX->framebufferPatches[i].find == nullptr || callbackIGFX->framebufferPatches[i].replace == nullptr)
+			continue;
+		
+		if (callbackIGFX->framebufferPatches[i].framebufferId != framebufferId)
+		{
+			framebufferAddress = callbackIGFX->findFramebufferId(callbackIGFX->framebufferPatches[i].framebufferId, (uint8_t *)callbackIGFX->gPlatformInformationList, framebufferSize);
+			framebufferId = callbackIGFX->framebufferPatches[i].framebufferId;
+		}
+		
+		KernelPatcher::LookupPatch patch;
+		
+		patch.kext = callbackIGFX->currentFramebuffer;
+		patch.find = (const uint8_t *)callbackIGFX->framebufferPatches[i].find->getBytesNoCopy();
+		patch.replace = (const uint8_t *)callbackIGFX->framebufferPatches[i].replace->getBytesNoCopy();
+		patch.size = min(callbackIGFX->framebufferPatches[i].find->getLength(), callbackIGFX->framebufferPatches[i].replace->getLength());
```

Comment:
It is better to assert that these are equal and skip the patch otherwise. No need to workaround user errors, at best we should SYSLOG and continue.